### PR TITLE
added turnaround time for merged pull requests

### LINKED ISSUE #143: Capture turn around time for PRs that are merged
When a PR is merged, the turnaround time for it is to be computed. It is the time taken from when the linked issue is assigned to the PR author to when it was merged.

### Tasks
- [ ] Compute and store the turn around time of merged PRs in the JSON data
- [ ] Show the turn around time for the PR event
- [ ] Show the avg. turn around time of a contributor in contributor profile

### PR DESCRIPTION
**Pull request by khavinshankar**
Fixes #143 

image

image

### CODE REVIEW

File: scraper/src/github.py [hunk 1]
```diff
@@ -198,10 +200,83 @@ def add_collaborations(self, event, event_time):
...
+        )
+        created_at = datetime.strptime(
+            event["payload"]["pull_request"]["created_at"], "%Y-%m-%dT%H:%M:%SZ"
+        )
+
+        linked_issues = re.findall(
+            r"(fix|fixes|fixed|close|closes|closed|resolve|resolves|resolved) ([\w\/.-]*)(#\d+)",
+            event["payload"]["pull_request"]["body"] or "",
+            re.IGNORECASE | re.MULTILINE,
+        )
```
Comment (rithviknishad) [inline scraper/src/github.py:220]: This wouldn't capture issues that are linked through the UI right?

image

File: app/contributors/[slug]/page.tsx [hunk 2]
```diff
@@ -264,6 +265,22 @@ export default async function Contributor({ params: { slug } }: Params) {
...
+              </dt>
+              <dd className="order-1 text-5xl font-extrabold text-foreground truncate whitespace-nowrap">
+                {formatDuration(
+                  (contributor.activityData?.activity
+                    .map((act) => act.turnaround_time)
+                    .filter(Boolean)
+                    .reduce(
+                      (acc, curr, i, array) => acc! + curr! / array.length,
+                      0
+                    ) || 0) * 1000
```
Comment (rithviknishad) [inline app/contributors/[slug]/page.tsx:280]: The averaging seems to be wrong here right?

Average shouldn't be dependent on the order of the elements right?

Your averaging technique is yielding different values when arr. is reversed.

Was weighted averaging intended?

image

Comment (khavinshankar) [reply app/contributors/[slug]/page.tsx:280]: forgot second param to the reduce

```js
arr.reduce((a, c, _, arr) => a + c / arr.length, 0)
```

image

**Comment by khavinshankar** [comment]
Closed this pr in favour of #155

**Pull request closed without merging**